ocaml/xapi/xapi_vmss.ml

```diff
@@ -90,20 +90,19 @@ let set_schedule ~__context ~self ~value =
 	Db.VMSS.set_schedule ~__context ~self ~value
 
 let set_type ~__context ~self ~value =
-	(* Check VMs associated to VMSS supports the new snapshot type *)
 	assert_licensed ~__context;
-	if value = `snapshot_with_quiesce then
+	(* For VMSS snapshot_type=snapshot_with_quiesce, Check VMs supports the snapshot_with_quiesce *)
+	if value = `snapshot_with_quiesce then begin
 		Pool_features.assert_enabled ~__context ~f:Features.VSS;
-	let snapshot_type = Record_util.vmss_type_to_string value in
-	let vms = Db.VMSS.get_VMs ~__context ~self in
-	if vms <> [] then
-		List.iter (fun vm ->
+		let snapshot_type = Record_util.vmss_type_to_string value in
+		Db.VMSS.get_VMs ~__context ~self
+		|> List.iter (fun vm ->
 			let allowed_operations = Db.VM.get_allowed_operations ~__context ~self:vm in
-			if not (List.exists (fun ty -> (Record_util.vm_operation_to_string ty) = snapshot_type)
-				allowed_operations) then
+			if not (List.exists (fun ty -> (Record_util.vm_operation_to_string ty) = snapshot_type) allowed_operations) then
```

Comment:
Is the conversion to strings for comparison necessary? The `snapshot` type is converted to a string as well as the `allowed_operation`. Is there no way to compare them before conversion?